### PR TITLE
add more test cases and code change to match blockchain changes

### PR DESCRIPTION
**Pull request by choonMy**

### CODE REVIEW

File: src.ts/non-fungible-token.ts [hunk 1]
```diff
@@ -269,7 +266,7 @@ export class NonFungibleToken {
             errors.throwError('accept fungible token ownership require signer', errors.NOT_INITIALIZED, { arg: 'signer' });
         }
 
-        this.isUsable; // check token useability, otherwise throw error
+         this.isUsable; // check token useability, otherwise throw error
```
Comment (yenkhoon) [inline src.ts/non-fungible-token.ts]: one extra space for indent

File: src.ts/utils/transaction.ts [hunk 2]
```diff
@@ -1193,37 +1193,107 @@ export function getTransactionRequest(route: string, transactionType: string, ov
                 };
             }
             break;
-        
+
```
Comment (yenkhoon) [inline src.ts/utils/transaction.ts]: remove extra line.

File: tests/test-0800-non-fungible-token-private.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ let wallet: mxw.Wallet;
 let provider: mxw.Wallet;
 let issuer: mxw.Wallet;
 let middleware: mxw.Wallet;
-let itemReceiver: mxw.Wallet;
+
```
Comment (yenkhoon) [inline tests/test-0800-non-fungible-token-private.ts]: remove extra line

File: tests/test-0810-non-fungible-token-public.ts [hunk 4]
```diff
@@ -24,18 +21,12 @@ let wallet: mxw.Wallet;
 let provider: mxw.Wallet;
 let issuer: mxw.Wallet;
 let middleware: mxw.Wallet;
-let itemReceiver: mxw.Wallet;
+
```
Comment (yenkhoon) [inline tests/test-0810-non-fungible-token-public.ts]: remove extra line

File: tests/test-0830-non-fungible-token-challenge.ts [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+        return mintedNFTItem.transfer(provider.address).then((receipt) => {
+            expect(receipt).to.not.exist;
+        }).catch(err => {
+            expect(err.code).to.equal(errors.CALL_EXCEPTION);
+        })
+    });
+
+});
+
+
```
Comment (yenkhoon) [inline tests/test-0830-non-fungible-token-challenge.ts]: remove extra line

File: tests/test-0830-non-fungible-token-challenge.ts [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+                return receipt;
+            }
+            return refresh(symbol).then(() => {
+                return receipt;
+            });
+        });
+    });
+}
+
+
```
Comment (yenkhoon) [inline tests/test-0830-non-fungible-token-challenge.ts]: remove extra line

File: src.ts/utils/transaction.ts [hunk 7]
```diff
@@ -1193,37 +1193,107 @@ export function getTransactionRequest(route: string, transactionType: string, ov
...
+        case "nonFungible/burnNonFungibleToken":
+            {
+                let params: {
+                    symbol: string,
+                    from: string,
+                    itemID: string,
+                    memo: string
+                } = checkFormat({
+                    symbol: checkString,
+                    from: checkString,
```
Comment (yenkhoon) [inline src.ts/utils/transaction.ts]: should be checkAddress

File: src.ts/utils/transaction.ts [hunk 8]
```diff
@@ -1193,37 +1193,107 @@ export function getTransactionRequest(route: string, transactionType: string, ov
...
+        case "nonFungible/updateNFTMetadata":
+            {
+                let params: {
+                    symbol: string,
+                    from: string,
+                    metadata: string,
+                    memo: string
+                } = checkFormat({
+                    symbol: checkString,
+                    from: checkString,
```
Comment (yenkhoon) [inline src.ts/utils/transaction.ts]: should be checkAddress

File: src.ts/utils/transaction.ts [hunk 9]
```diff
@@ -1193,37 +1193,107 @@ export function getTransactionRequest(route: string, transactionType: string, ov
...
-                                        from: params.from,
-                                    }
+            {
+                let params: {
+                    symbol: string,
+                    from: string,
+                    memo: string
+                } = checkFormat({
+                    symbol: checkString,
+                    from: checkString,
```
Comment (yenkhoon) [inline src.ts/utils/transaction.ts]: should be checkAddress

**Comment by yenkhoon** [comment]
we will need to have the reject token creation test cases and also after transfer item, transfer again.

**Comment by b00f** [comment]
> we will need to have the reject token creation test cases and also after transfer item, transfer again.

@jeansoonsik please make sure you have covered these cases.

File: tests/test-0800-non-fungible-token-private.ts [hunk 10]
```diff
@@ -194,7 +188,7 @@ describe('Suite: NonFungibleToken - private', function () {
                     expect(receipt).is.not.exist;
                 }).catch(error => {
                     expect(error.code).to.equal(errors.NOT_ALLOWED);
-                });
+                }); true
```
Comment (jeansoonsik) [inline tests/test-0800-non-fungible-token-private.ts]: Extra stuff

**Pull request merged**